typescript/keys.d.ts

```diff
@@ -26,7 +26,15 @@ export interface NewPowerAutomateKey extends BaseKey {
   type: 'POWER_AUTOMATE'
 }
 
-export type NewKey = NewLegacyKey | NewDeveloperKey | NewPowerAutomateKey
+export interface NewSubmissionDataKey extends BaseKey {
+  type: 'SUBMISSION_DATA'
```

Comment:
```suggestion
  type: 'SUBMISSION_DATA'
  privilege: {}
```

I think this will make it easier to add more config later and not have to make the `privilege` column nullable in our database

Reply:
`{}` is prob not the right way to do this.